### PR TITLE
Update Protocol tests to use Settings object

### PR DESCRIPTION
**Pull request by richardjgowers**
Updates the tests around `DummyProtocol` to use a `Settings` object internally, making the tests more relevant for real world cases

### CODE REVIEW

**Comment by mikemhenry** [comment]
Really weird that only RTD check shows up here, but the CI action and the mypy action ran and passed

**Comment by richardjgowers** [comment]
@mikemhenry might be because I was PRing onto a different branch than main?

**Comment by mikemhenry** [comment]
Yes! That was it, we have it (correctly) set up to only run on PRs pointed at main. https://github.com/OpenFreeEnergy/gufe/blob/main/.github/workflows/ci.yaml#L5

**Comment by dotsdl** [comment]
Reviewing!

File: gufe/protocols/protocol.py [hunk 1]
```diff
@@ -73,9 +73,9 @@ def get_rate_of_convergence(self):
 class Protocol(GufeTokenizable):
     """A protocol that implements an alchemical transformation.
 
-    Takes a `ProtocolSettings` object specific to the protocol on init.
-    This configures the protocol for repeated execution on `ChemicalSystem`
-    objects.
+    Takes a `Settings` object customised for this protocol on init.
```
Comment (dwhswenson) [inline gufe/protocols/protocol.py:76]:
```suggestion
    Takes a ``Settings`` object customised for this protocol on init.
```

**Review by dwhswenson — APPROVED** [review]
Suggested some md -> rst fixes in the docstrings; otherwise LGTM.

I'm assuming updating to main will fix the CodeCov project complaints.

Also, note that `openfecli` re-uses `DummyProtocol` from here, so we'll likely need to update the saved JSON files there. There's a script to do that in `openfecli/tests/dev`.

**Review by dotsdl — APPROVED** [review]
It's a yes from me dawg.

File: gufe/protocols/protocol.py [hunk 2]
```diff
@@ -91,23 +91,22 @@ class Protocol(GufeTokenizable):
         Corresponding `ProtocolResult` subclass
 
     """
-
+    _settings: Settings
     result_cls: type[ProtocolResult]
 
-    def __init__(self, settings=None):
-        """Create a new `Protocol` instance.
+    def __init__(self, settings: Settings):
```
Comment (richardjgowers) [inline gufe/protocols/protocol.py:97]: `Settings` now not optional, I couldn't think of a case (besides lazy test writing, which I'm fixing here) when we'd have a Settings-less Protocol

File: gufe/settings/models.py [hunk 3]
```diff
@@ -85,7 +87,7 @@ class Settings(SettingsBaseModel):
     settings_version: int = 0
     forcefield_settings: OpenMMSystemGeneratorFFSettings
     thermo_settings: ThermoSettings
-    protocol_settings: Union[ProtocolSettings, None]
+    protocol_settings: type[ProtocolSettings]
```
Comment (richardjgowers) [inline gufe/settings/models.py:90]: Also `Settings` must include the Protocol specific section.  Again, I couldn't think of a good example of when you'd not want to have Protocol specific settings.  If there was one, you could always define an empty dict.

**Comment by richardjgowers** [comment]
@dwhswenson @dotsdl I took it all a bit further, sorry

**Pull request merged**